dotCMS/src/integration-test/java/com/dotcms/publisher/bundle/business/BundleAPITest.java

```diff
@@ -0,0 +1,205 @@
+import java.util.Date;
+import java.util.Set;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BundleAPITest {
+
+    private static User adminUser;
+    private static BundleAPI bundleAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+        adminUser = APILocator.systemUser();
+        bundleAPI = APILocator.getBundleAPI();
+    }
+
+    private String insertPublishingBundle(final String userId, final Date publishDate)
+            throws DotDataException {
+        final String uuid = UUIDGenerator.generateUuid();
+        final Bundle bundle = new Bundle();
+        bundle.setId(uuid);
+        bundle.setName("testBundle"+System.currentTimeMillis());
+        bundle.setForcePush(false);
+        bundle.setOwner(userId);
+        bundle.setPublishDate(publishDate);
+        APILocator.getBundleAPI().saveBundle(bundle);
+
+        return uuid;
+    }
+
+    private void insertPublishAuditStatus(final Status status, final String bundleID) throws DotPublisherException {
+        final PublishAuditStatus publishAuditStatus = new PublishAuditStatus(bundleID);
+        publishAuditStatus.setStatusPojo(new PublishAuditHistory());
+        publishAuditStatus.setStatus(status);
+        APILocator.getPublishAuditAPI().insertPublishAuditStatus(publishAuditStatus);
+    }
+
+    @Test
+    public void test_deleteBundleAndDependencies_byAdmin() throws DotDataException {
+        final User newUser = new UserDataGen().nextPersisted();
+        final String bundleIdAdmin = insertPublishingBundle(adminUser.getUserId(),new Date());
+        final String bundleIdUser = insertPublishingBundle(newUser.getUserId(),new Date());
+
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser));
+
+        bundleAPI.deleteBundleAndDependencies(bundleIdAdmin,adminUser);
+        bundleAPI.deleteBundleAndDependencies(bundleIdUser,adminUser);
+
+        assertNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNull(bundleAPI.getBundleById(bundleIdUser));
+    }
+
+    @Test(expected = DotDataException.class)
+    public void test_deleteBundleAndDependencies_byLimitedUser() throws DotDataException {
+        final User newUser = new UserDataGen().nextPersisted();
+        final String bundleIdAdmin = insertPublishingBundle(adminUser.getUserId(),new Date());
+        final String bundleIdUser = insertPublishingBundle(newUser.getUserId(),new Date());
+
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser));
+
+        bundleAPI.deleteBundleAndDependencies(bundleIdUser,newUser);
+        assertNull(bundleAPI.getBundleById(bundleIdUser));
+
+        bundleAPI.deleteBundleAndDependencies(bundleIdAdmin,newUser);
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
```

Comment:
this line should never happen according to the expected, why having it?

Reply:
Done